tempto-core/src/main/java/com/teradata/tempto/internal/fulfillment/table/hive/HiveThriftClient.java

```diff
@@ -132,16 +133,15 @@ private ColumnStatisticsObj toHiveColumnStatistics(FieldSchema fieldSchema, Colu
         ColumnStatisticsData hiveColumnStatisticsData = new ColumnStatisticsData();
         String type = fieldSchema.getType();
         if (type.startsWith("varchar") || type.startsWith("char")) {
-            // length is stored as something like: varchar(12)
-            int maxLength = Integer.parseInt(type.substring(type.indexOf("(") + 1).replaceAll("\\)", ""));
+            int maxLength = parseInt(type.substring(type.indexOf("(") + 1).replaceAll("\\)", ""));
             StringColumnStatsData hiveColumnStatsData = new StringColumnStatsData();
             hiveColumnStatsData.setNumDVs(columnStatistics.getDistinctValuesCount());
             hiveColumnStatsData.setMaxColLen(maxLength);
             hiveColumnStatsData.setAvgColLen(maxLength / 3.0);
             hiveColumnStatsData.setNumNulls(columnStatistics.getNullsCount());
             hiveColumnStatisticsData.setStringStats(hiveColumnStatsData);
         }
-        else if (type.equals("bigint") || type.equals("int")) {
+        else if (type.equals("tinyint") || type.equals("smallint") || type.equals("int") || type.equals("bigint")) {
```

Comment:
I saw it always in lower case. So it is hard to tell if it is by design or if it is just a coincidence (because tempto created these tables).